src/gui/dialogs/end_credits.hpp

```diff
@@ -47,16 +47,28 @@ class end_credits : public modal_dialog
 	void timer_callback();
 	void key_press_callback(const SDL_Keycode key);
 
+	
 	const std::string& focus_on_;
 
 	std::vector<std::string> backgrounds_;
-
 	scroll_label* text_widget_;
 
 	// The speed of auto-scrolling, specified as px/s
 	int scroll_speed_;
 
 	uint32_t last_scroll_;
+	//Fix credits crash 
+	static constexpr size_t sliding_size_=2; // slidingSize alters how many of the sliding contents are to be run at once 
+						  // n-1 = 2 => 3 strings at once concatinated 
+	static constexpr size_t screen_space_ = 300; // magic number #1  
+	static constexpr size_t magic_number_ = 150; // magic number #2
```

Comment:
This definitely needs a better name.

Also generally, it would be better for these variable comments to be in doxygen format.